doc/manual/metaprogramming.rst

```diff
@@ -9,106 +9,120 @@
 The strongest legacy of Lisp in the Julia language is its metaprogramming
 support. Like Lisp, Julia represents its own code as a data structure of
 the language itself.
+Because all data types and code in Julia are represented by Julia data structures,
+powerful reflection capabilities are available to explore the internals of a
+program and its types just like any other data.
 Since code is represented by objects that can be created and manipulated
 from within the language, it is possible for a program to transform and
 generate its own code. This allows sophisticated code generation without
-extra build steps, and also allows true Lisp-style macros, as compared
-to preprocessor "macro" systems, like that of C and C++, that perform
-superficial textual manipulation as a separate pass before any real
-parsing or interpretation occurs. Another aspect of metaprogramming is
-reflection: the ability of a running program to dynamically discover
-properties of itself. Reflection emerges naturally from the fact that
-all data types and code are represented by normal Julia data structures,
-so the structure of the program and its types can be explored
-programmatically just like any other data.
-
-Expressions and :func:`eval`
-----------------------------
-
-Julia code is represented as a syntax tree built out of Julia data
-structures of type :obj:`Expr`. This makes it easy to construct and
-manipulate Julia code from within Julia, without generating or parsing
-source text. Here is the definition of the :obj:`Expr` type::
-
-    type Expr
-      head::Symbol
-      args::Array{Any,1}
-      typ
-    end
+extra build steps, and also allows true Lisp-style macros operating at
+the level of abstract syntax. In contrast, preprocessor "macro" systems,
```

Comment:
@jakebolewski done

@ tlycken I think "Lisp-style" is descriptive for people who understand what it means and reasonably neutral otherwise (i.e. "keep reading to find out")